src/wfs/src/main/java/org/geoserver/wfs/json/ComplexGeoJsonWriter.java

```diff
@@ -416,7 +417,10 @@ private Object getSimpleContent(ComplexAttribute property) {
             return null;
         }
         Object value = simpleContent.getValue();
-        if (value instanceof Number || value instanceof String || value instanceof Character) {
+        if (value instanceof Number
```

Comment:
Wondering if this check is rely needed or if we could just return whatever is the simple content value? But I agree that if it is not clear then let's keep it, feels safer.

Reply:
Indeed I did not touch cause I thought the existing if was there for a reason (But I don't know which one, ha!)